smac/smbo/smbo.py

```diff
@@ -126,12 +127,16 @@ def run(self):
 
             self.logger.debug("Intensify")
 
-            self.incumbent, inc_perf = self.intensifier.intensify(
-                challengers=challengers,
-                incumbent=self.incumbent,
-                run_history=self.runhistory,
-                aggregate_func=self.aggregate_func,
-                time_bound=max(0.01, time_spend))
+            try: self.incumbent, inc_perf = self.intensifier.intensify(
+                  challengers=challengers,
+                  incumbent=self.incumbent,
+                  run_history=self.runhistory,
+                  aggregate_func=self.aggregate_func,
+                  time_bound=max(0.01, time_spend))
+            except (TAEAbortException, BudgetExhaustedException):
```

Comment:
I chose the minimal-invasive solution and introduced an intensifier.incumbent_on_error that is set through exception-handling. Also added another try/except around the initial-run in smbo.